docs/design/live_migration.md

```diff
@@ -0,0 +1,176 @@
+
+┌─────────────────────────────────┐
+│ dhcp-options 1234               │
+│   lease_time: 3500              │
+│   router: 169.254.1.1           │
+│   dns_server: [kubedns]         │
+│   server_id: 169.254.1.1        │
+│   server_mac: c0:ff:ee:00:00:01 │
+└─────────────────────────────────┘
+```
+
+**Virt-launcher pod address:**
+The CNI will not set an address at virt-launcher pod netns, that address is
+assigned to the VM with the DHCP options from the LSP, this allows to use 
+kubevirt bridge binding with pod networking and still do live migration.
+
+#### IPAM
+
+The point to point routing feature allows an address to be running at a node different from
+the one "owning" the subnet the address is coming from. This will happen
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what IP pool the VM belongs.
+
+To cover all this scenarios the VM's pods are labeled with `k8s.ovn.org/original-switch-name`
+with the switch name where the VM was created since is the key for the IP pool to allocate IP, 
+this label is copied during live migration to the new virt-launcher pod runnig
+at the migration target node
+
+
+#### Detecting migratable VMs and changing point to point routes
+
+To detect that a pod is migratable KubevirtVm the annotation `kubevirt.io/allow-pod-bridge-network-live-migration`
+has to be present and also the label `vm.kubevirt.io/name=vm1`, on that case
+the point to point routes will be created after live migration and also a 
+DHCPOptions will be configured to serve the ip configuration to the VM
+
+During live migration there is source and target virt-launcher pod's with different names, the ovn-k pod's controller use `CreationTimestamp` and `kubevirt.io/vm=vm1`
+to differentiate between them, then it watch the following annotation and label 
+at target pod `kubevirt.io/nodeName` and `kubevirt.io/migration-target-start-timestamp`:
+- The `kubevirt.io/nodeName` is set after the VM finishes live migrating or when it becomes ready.
+- The `kubevirt.io/migration-target-start-timestamp` is set when live migration has not finished but migration-target pod is ready to receive traffic (this happens at post-copy live migration, where migration is taking too long).
+
+The point to point routing cleanup (remove of static route and policy) will be done at two cases:
+- VM is deleted.
+- VM is live migrated back to the node that owns its IP.
+- ovn-kubernetes controllers are restarted (sync virtual machines)
```

Comment:
I don't understand this item; it I restart the ovnkube master, will I remove the routes / policies for the affected nodes? The way it is written makes me believe that is what happens.

Reply:
It will remove "stale", I will add that to the last bullet.